BrainPortal/lib/globus_helpers.rb

```diff
@@ -75,161 +64,143 @@ def globus_fetch_token(code, globus_action_url)
   end
 
-  def set_of_identity_provider_names(globus_identity)
-    set_of_identities(globus_identity).map { |s| s['identity_provider_display_name'] }
-  end
-
   # Returns an array of allowed identity provider names.
   # Returns nil if they are all allowed
-  def allowed_globus_provider_names(user)
+  # Keep reference to globus for backward compability
+  def allowed_oidc_provider_names(user)
     user.meta[:allowed_globus_provider_names]
-       .presence
-      &.split(/\s*,\s*/)
-      &.map(&:strip)
+      .presence
+     &.split(/\s*,\s*/)
+     &.map(&:strip)
   end
 
-  def user_can_link_to_globus_identity?(user, identity)
-    allowed = allowed_globus_provider_names(user)
+  def user_can_link_to_oidc_identity?(oidc, user, identity) #:nodoc:
+    allowed = allowed_oidc_provider_names(user)
+
     return true if allowed.nil?
     return true if allowed.size == 1 && allowed[0] == '*'
-    prov_names = set_of_identity_provider_names(identity)
-    return true if (allowed & prov_names).present? # if the intersection is not empty
+    return true if allowed.include?(identity[oidc.identity_provider_display_name_key])
+
     false
   end
 
-  def user_has_link_to_globus?(user)
-    user.meta[:globus_provider_id].present?       &&
-    user.meta[:globus_provider_name].present?     &&
-    user.meta[:globus_preferred_username].present?
+  def user_has_link_to_oidc?(user) #:nodoc:
+    # Filter out the identities that are not allowed
+    allowed_oidc_names = allowed_oidc_provider_names(user)
+    oidc_providers     = OidcConfig.enabled
+
+    # Iterate over the allowed_oidc_info
+    oidc_providers.any? do |oidc|
+      oidc.linked_provider_id(user).present? &&
+      oidc.linked_provider_name(user).present? &&
+      oidc.linked_preferred_username(user).present? && 
+      ( allowed_oidc_names.include?('*') ||  
+        allowed_oidc_names.include?(oidc.linked_provider_name(user))
+      )
+    end
   end
 
-  def user_must_link_to_globus?(user)
+  def user_must_link_to_oidc?(user)
     user.meta[:allowed_globus_provider_names].present?
   end
 
-  def wipe_user_password_after_globus_link(user)
-    user.update_attribute(:crypted_password, 'Wiped-By-Globus-Link-' + User.random_string)
-    user.update_attribute(:salt            , 'Wiped-By-Globus-Link-' + User.random_string)
+  def wipe_user_password_after_oidc_link(oidc, user)
+    user.update_attribute(:crypted_password, "Wiped-By-#{oidc.name}-Link-" + User.random_string)
+    user.update_attribute(:salt            , "Wiped-By-#{oidc.name}-Link-" + User.random_string)
     user.update_attribute(:password_reset  , false)
   end
 
-  # Given a globus identity structure, find the user that matches it.
-  # Returns the user object if found; returns a string error message otherwise.
-  def find_user_with_globus_identity(globus_identity)
+  def find_user_with_oidc_identity(oidc, identity)
+    provider_name = identity[oidc.identity_provider_display_name]
```

Comment:
Again, use `oidc.identity_info()` ; the code here crashes because you did not update the method names to say 'NNN_key'. This would not have happend when using `identity_info()`